lifecycle-hooks.md

```diff
@@ -0,0 +1,95 @@
+**Table of Contents**
+- [Using Cluster Lifecycle Hooks](#using-cluster-lifecycle-hooks)
+  - [Running the Extension](#running-the-extension)
+  - [Register the Extension](#register-the-extension)
+  - [Extension in Action](#extension-in-action)
+
+# Using Cluster Lifecycle Hooks
+
+This guide covers how one can use Cluster API's RuntimeSDK feature to hook into key lifecycle events of the Cluster. 
+In this section we will create a simple `test-extension` that receives key lifecycle events of the cluster and prints them.
+
+## Running the Extension
+
+Firstly we will need a simple extension server that needs to run and be able to receive cluster lifecycle events. For this, lets create a deployment in the management cluster that will act as our extension server.
+```
+kubectl apply -f yamls/extension/test-extension-deployment.yaml
+```
+
+Note that the YAML also contains the all the necessary resources like Service, Certificates, etc that will make this extension server reachable from within the management cluster.
+
+Let's verify that the extension server is running:
+```
+kubectl get deployments -n test-extension-system
+```
+The output should resemble:
+```
+NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
+test-extension-manager   1/1     1            1           30s
+```
+
+## Register the Extension
+
+Now that we have a running extension server let's register it with ClusterAPI. This will inform ClusterAPI to call the extension server during key workload cluster lifecycle events.
```

Comment:
```suggestion
Now that we have a running extension server let's register it with Cluster API. This will inform Cluster API to call the extension server during key workload cluster lifecycle events.
```
nit (everywhere)